extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleJobStore.java

```diff
@@ -237,7 +238,7 @@ public UUID findFirst(JobAuthorization.State jobState) {
   }
 
   @Override
-  public <T extends DataModel> void create(UUID jobId, String key, T model) throws IOException {
```

Comment:
i'd prefer to avoid changing the interface here so we know only DataModels are still being stored in JobStore.

If the executor interface requires serializable, can you convert to DataModel inside?

Reply:
Why? Specifically why do we care what data is stored in job store? Limiting it to just DataModels prevents us from storing all kind of useful things like simple types and types from sdks.

Reply:
So I guess it's a difference between a very open API contract with undefined objects or more well-defined objects to store.

As far as tradeoffs: 
-If we define the contract narrowly using DataModel, are we doing unnecessary object conversions to fit the model?
- If we allow any Serializable object, are we open to misuse or less optimal use?

I don't mind at this point. My inclination is to leave it well-defined until we see a use case where it's a hindrance.

Reply:
ah ok, sounds like there are arguments for opening it up, just wanted to flag that it was a change from previous behavior that I think was there for a reason. I remember there being some earlier discussions with Jim about this but do not recall the details / history of the design.

Reply:
So for what it's worth it is a hindrance now.  I'll make the change back just so we can get this in and then evaluate the change to serializable on its own but examples where it is a problem:

1) Existing behavior with temp data objects being created and stored one per job is a huge anti-pattern that was created because you couldn't store simple data in the data store.

2) This current change wouldn't be able to use the JobStore easily without wrapping everything in a generic wrapper, which I guess I can do but that seems really goony.